VirtoCommerce.Storefront.Model/Common/ValueObject.cs

```diff
@@ -59,17 +62,19 @@ protected virtual IEnumerable<object> GetEqualityComponents()
                 var value = property.GetValue(this);
                 if (value == null)
                 {
-                    yield return "null";
+                    yield return null;
```

Comment:
it was important to return "null" as a string if the value not set to preserve properties order when a cache key constructing. The `string.Join`  will skip null values and the resulting cache key may contain a various number of key parts.

Reply:
`null` and `"null"` are different values. When you replace one with another you get the same equality components for different values and you get duplicate cache keys. This PR fixes exactly this bug.

Reply:
As for _preserving  properties order_ - this order is broken when you just skip empty collections, and two objects
`{ List1 = { "A" }, List2 = { } }`
`{ List1 = { }, List2 = { "A" } }`
become equal and produce the same cache key. This bug is also fixed in this PR.

Reply:
All these cases are covered by tests.

Reply:
BTW, `string.Join()` doesn't skip `null` values. `string.Join("|", null, null, null)` returns `||`.

Reply:
Could you do the same PR for the platform?
https://github.com/VirtoCommerce/vc-platform/blob/master/src/VirtoCommerce.Platform.Core/Common/ValueObject.cs

Reply:
Sure